libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -231,6 +231,20 @@ impl NotificationHandler {
     }
 }
 
+impl Iterator for NotificationHandler {
+    type Item = CmNotification;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.recv.recv() {
+            Ok(notification) => Some(notification),
+            Err(_) => {
+                error!("connection manager no longer running");
```

Comment:
Is this really an error?  This message will be displayed on a clean shutdown.  Probably should be debug message.